pkg/server/plugin/upstreamca/spire/spire_workload.go

```diff
@@ -0,0 +1,113 @@
+package spireplugin
+
+import (
+	"context"
+	"crypto/x509"
+	"net"
+	"time"
+
+	"encoding/pem"
+	"github.com/spiffe/spire/api/workload"
+	proto "github.com/spiffe/spire/proto/api/workload"
+	"io/ioutil"
+	"os"
+)
+
+func (m *spirePlugin) getWorkloadSVID(ctx context.Context, config *Configuration) ([]byte, []byte, []byte, string, error) {
+	errorChan := make(chan error, 1)
+
+	duration, _ := time.ParseDuration(config.TTL)
+	wapiClient := m.newWorkloadAPIClient(config.ServerAgentAddr, duration)
+	updateChan := wapiClient.UpdateChan()
+	go func() {
+		err := wapiClient.Start()
+		if err != nil {
+			errorChan <- err
+		}
+	}()
+
+	defer wapiClient.Stop()
+
+	for {
+		select {
+		case svidResponse := <-updateChan:
+			return m.receiveUpdatedCerts(svidResponse)
+		case <-ctx.Done():
+			return []byte{}, []byte{}, []byte{}, "", nil
+		case err := <-errorChan:
+			return []byte{}, []byte{}, []byte{}, "", err
+		}
+	}
+}
+
+func (m *spirePlugin) debugDumpCerts(certificate []byte, key []byte, bundle []byte) error {
```

Comment:
Is this used anywhere? Looks like it can be removed, along with the write* functions below

Reply:
Stray debug functions , removed them in recent commits.